### PR TITLE
[ign-common3] Support fuel URLs for textures

### PR DESCRIPTION
**Pull request by JShep1**
Resolves https://github.com/ignitionrobotics/ign-gazebo/issues/343

Signed-off-by: John Shepherd <john@openrobotics.org>

### CODE REVIEW

**Review by mjcarroll — COMMENTED** [review]
Are there any tests you could add here for this?

**Comment by JShep1** [comment]
> Are there any tests you could add here for this?

Added in https://github.com/ignitionrobotics/ign-common/pull/102/commits/7b02ea6b4f9201b718c526bfd6358bf9ad80c1b6

**Comment by mjcarroll** [comment]
Looks like your new test is failing CI.

File: graphics/src/Material.cc [hunk 1]
```diff
@@ -149,11 +149,17 @@ void Material::SetTextureImage(const std::string &_tex,
...
   {
-    this->dataPtr->texImage = _resourcePath + "/../materials/textures/" + _tex;
+    // Try to resolve this texture image to a locally cached path in a
+    // separate directory
+    this->dataPtr->texImage = common::findFile(_tex);
     if (!exists(this->dataPtr->texImage))
     {
-      ignerr << "Unable to find texture[" << _tex << "] in path["
-            << _resourcePath << "]\n";
+      this->dataPtr->texImage = _resourcePath + "/../materials/textures/" + _tex;
```
Comment (mjcarroll) [inline graphics/src/Material.cc:157]: Make linter happy

```suggestion
      this->dataPtr->texImage = _resourcePath + "/../materials/textures/" 
          + _tex;
```

File: graphics/src/Material_TEST.cc [hunk 2]
```diff
@@ -38,6 +39,12 @@ TEST_F(MaterialTest, Material)
   EXPECT_STREQ("/path/../materials/textures/texture_image",
                mat.TextureImage().c_str());
 
+  std::string filenameIn = std::string(PROJECT_SOURCE_PATH) +
```
Comment (mjcarroll) [inline graphics/src/Material_TEST.cc:42]: We should probably use `ignition::common::jointPaths` here.

File: graphics/src/Material.cc [hunk 3]
```diff
@@ -149,11 +149,18 @@ void Material::SetTextureImage(const std::string &_tex,
...
   {
-    this->dataPtr->texImage = _resourcePath + "/../materials/textures/" + _tex;
+    // Try to resolve this texture image to a locally cached path in a
+    // separate directory
+    this->dataPtr->texImage = common::findFile(_tex);
     if (!exists(this->dataPtr->texImage))
     {
-      ignerr << "Unable to find texture[" << _tex << "] in path["
-            << _resourcePath << "]\n";
+      this->dataPtr->texImage = _resourcePath + "/../materials/textures/"
```
Comment (mjcarroll) [inline graphics/src/Material.cc:157]: Prefer `joinPaths`

**Review by mjcarroll — COMMENTED** [review]
Sorry, one more thing, we should be using `joinPaths` where possible.

The Windows CI failure is flaky test, I believe, as it is unrelated to this change.

File: graphics/src/Material_TEST.cc (hunk 2 above)
Comment (mjcarroll) [reply graphics/src/Material_TEST.cc:42]: @JShep1 ping here too

**Review by mjcarroll — APPROVED** [review]
LGTM with green CI

File: graphics/src/Material_TEST.cc (hunk 2 above)
Comment (JShep1) [reply graphics/src/Material_TEST.cc]: You might want to double check my update to the tests, specifically changing `/path` to` `path`, does common have a way of specifying root, prepending `/` before any sort of path?

**Comment by JShep1** [comment]
> LGTM with green CI

Replied on one of your comments but commenting here too in case it gets swallowed:  You might want to double check my updates to the tests, specifically for `/path` being changed to `path`, is there a way of specifying root, prepending `/` with `common`?

**Comment by JShep1** [comment]
@osrf-jenkins run tests please

**Comment by mjcarroll** [comment]
> You might want to double check my updates to the tests, specifically for /path being changed to path, is there a way of specifying root, prepending / with common?

You should be able to do `joinPaths("/path", "foo", "bar", "baz")`

**Comment by mjcarroll** [comment]
ignition_common-ci-pr_any-ubuntu_auto-amd64  and ignition_common-abichecker-any_to_any-ubuntu_auto-amd64 were infra issues.  I think this is probably safe to merge.

**Comment by JShep1** [comment]
@osrf-jenkins run tests please

**Comment by mjcarroll** [comment]
Infra issues again, I'm going to go ahead and merge.

**Pull request merged**